### PR TITLE
Generate provenance statements on npm publish

### PR DESCRIPTION
**Pull request by wojtekmaj**
This PR adds [Provenance statements](https://docs.npmjs.com/generating-provenance-statements) on `npm publish`, increasing supply-chain security.

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
*As a general rule:* Please note that the context provided in https://github.com/mozilla/pdf.js/pull/18352#issue-2383563292 should also be included in the *commit message* itself such that it's possible to understand what the patch does *and importantly why* on the Git command line without having to reference the GitHub PR.

File: .github/workflows/publish_release.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 name: Publish release
+
```
Comment (timvandermeij) [inline .github/workflows/publish_release.yml:2]: Please remove this newline change and the one below. It's not really related to the patch itself and the other workflows also have no newlines here. (If we want that, it should be done consistently in all workflows in a separate PR.)

**Review by timvandermeij — APPROVED** [review]
Looks good to me, with the comment addressed in the squashed commit. In the meantime I'll read up on provenance a bit more before merging to make sure all prerequisites are met (because I'd like to make a new release later today).

File: .github/workflows/publish_release.yml (hunk 1 above)
Comment (wojtekmaj) [reply .github/workflows/publish_release.yml]: Fixed, thanks!

**Comment by timvandermeij** [comment]
Thanks!

**Comment by wojtekmaj** [comment]
Uh. So I guess repository.url in package.json must match the URL of the repository GitHub Actions are ran from. In other words, it cannot point to https://github.com/mozilla/pdfjs-dist, but to https://github.com/mozilla/pdf.js for provenance to work.

**Comment by timvandermeij** [comment]
@wojtekmaj Sadly this failed in GitHub Actions for the new release; please see https://github.com/mozilla/pdf.js/actions/runs/9748332722/job/26903097065#step:6:359. The issue appears to be that in https://github.com/mozilla/pdf.js/commit/dddb74d5e4d422b2be80ee93c26396830be2cfb3 we normalized the repository URL according to NPM recommendation, and I noticed you have the same in `react-pdf` at https://github.com/wojtekmaj/react-pdf/blob/main/packages/react-pdf/package.json#L105, so I'm wondering if you also ran into this issue and how you fixed it?

**Comment by wojtekmaj** [comment]
Commented split second before you, so commenting now just in case the notification doesn't get to you :D

**Comment by timvandermeij** [comment]
Ha, thanks. I guess the best way forward is to unpublish this release, fix the URL and try again? Fortunately the error caused nothing to be published to NPM yet AFAICT, so I think unpublishing should be safe.

**Comment by wojtekmaj** [comment]
Both options should work!

**Comment by timvandermeij** [comment]
Thanks; the issue should be fixed by https://github.com/mozilla/pdf.js/pull/18358.

**Comment by timvandermeij** [comment]
It worked, and the provenance badge is now shown on https://www.npmjs.com/package/pdfjs-dist/v/4.4.168. Nice!
